### PR TITLE
groups: creation using a profile and some reflection

### PR DESCRIPTION
**Pull request by greut**
Addition of `CreateSecurityGroup` using a `SecurityGroupProfile`. We are bound to document the whole CloudStack API in Go structs... but it already a little bit nicer.

And super usage of the `reflect` package to create POST requests based on a `struct`

### CODE REVIEW

**Comment by greut** [comment]
List of the supported calls so far... https://github.com/exoscale/egoscale/wiki/api

File: request.go [hunk 1]
```diff
@@ -190,3 +206,85 @@ func (exo *Client) DetailedRequest(uri string, params string, method string, hea
...
+
+// prepareValues uses a profile to build a POST request
+func prepareValues(profile interface{}) (*url.Values, error) {
+	params := url.Values{}
+	value := reflect.ValueOf(profile)
+	typeof := reflect.TypeOf(profile)
+	for typeof.Kind() == reflect.Ptr {
+		typeof = typeof.Elem()
+		value = value.Elem()
+	}
```
Comment (marcaurele) [inline request.go]: For my know-how, what is the intent of this loop?

Comment (greut) [reply request.go]: `interface{}` could be the `struct` I want to work on, but also a `*struct`, or a `**struct`, and so on.

**Comment by greut** [comment]
too big to parse

**Pull request closed without merging**